adding test_msgs to rcl_interfaces repo

mikaelarguedas opened the pull request:
replaces https://github.com/ros2/common_interfaces/pull/30
Fix ros2/system_tests#191
Connects to ros2/system_tests#191

dirk-thomas committed:
add generic tests for pub/sub and req/rep, add two message and service types for now

dirk-thomas committed:
add test for messages with static arrays

dirk-thomas committed:
add test for messages with dynamic arrays

dirk-thomas committed:
use double curly braces on vector init lists to work on Windows

dirk-thomas committed:
add tests for nested messages

dirk-thomas committed:
add tests for builtin messages

dirk-thomas committed:
refactor message generation (ros2/ros2#48)

wjwwood committed:
adjust use of braces to fix warnings with clang

wjwwood committed:
fix narrowing conversion error on windows

dirk-thomas committed:
update message API

wjwwood committed:
changes to support renaming of interface packages

dirk-thomas committed:
add test to cover messages with an array of sub messages

dirk-thomas committed:
remove package name prefix

dirk-thomas committed:
add test to cover messages with a static array of sub messages

dirk-thomas committed:
add fixtures with a string with more then 255 characted and more than 100 elements in a sequence

dirk-thomas committed:
fix more windows warnings

dirk-thomas committed:
use linters

dirk-thomas committed:
update code to pass ament_cpplint

dirk-thomas committed:
extend test to cover ros2/rclcpp#219

mikaelarguedas committed:
working towards testing python communication along cpp one

Conflicts:
	test_communication/test/message_fixtures.hpp

mikaelarguedas committed:
added back char and byte now that somehow fixed on python generator side

Conflicts:
	test_communication/test/message_fixtures.hpp

dirk-thomas committed:
fix spelling in comment

dirk-thomas committed:
add communication tests for bounded arrays

mikaelarguedas committed:
use values that fit in a float for testing

mikaelarguedas committed:
fix brackets: see if Clang or Windows complains

mikaelarguedas committed:
use utf-8 compatible values for char testing

mikaelarguedas committed:
add python message fixtures

mikaelarguedas committed:
testing values cross rcl communication

mikaelarguedas committed:
use camelcase format for message names

mikaelarguedas committed:
cleanup

dirk-thomas committed:
support bounded vectors

mikaelarguedas committed:
Ascii fixture (#161)

* use ASCII value for char

* homogenize BoundedArrayPrimitives fixtures

dirk-thomas committed:
comply with stricter cpplint rules

dirk-thomas committed:
tests strings > 256

mikaelarguedas committed:
Test python services (#175)

* extend service template to test python services

* trailing whitespace

* skipping tests raising SkipTest

* remove SKIP_TEST for non nose tests

* add bracket because linter doesnt understand multiline conditions

mikaelarguedas committed:
test loong strings for services (#179)

dirk-thomas commented:
Once the message definitions and fixtures have been move https://github.com/ros2/ros2cli/blob/9fce6a08381bb67cf17e7160b0b2c6556ffec1e6/ros2topic/package.xml#L16 should switch to the new location.

mikaelarguedas committed:
C memleak testing (#211)

* Added nested message that always breaks because of the bug

* Added C++ code for DynamicArrayPrimitivesNested message

* Fixed style and publisher/subscriber (combo) test case

* expose core dumpes on complex messages

* dont run other tests to save debugging time

* more fixtures, looks like a string array alignment issue

* newline at end of file

* move include to the right place

* add comment about current failing tests

* remove debug prints

* restore/reenable all tests

* that was actually pretty readable with vertical space

* use all messages fron the fixtures rather the only the first one

* linters

* what's cool with functions is that you can call them rather than copy-n-paste code

mikaelarguedas committed:
testing array longers than 101 (#216)

dirk-thomas commented:
I think it would be valuable to preserve the history of the moved content. These files have quite some history which would be valuable for future debugging / tracing.

mikaelarguedas committed:
move fixtures to subfolder

mikaelarguedas committed:
add CMakeLists and manifest

add __init__.py

mikaelarguedas committed:
update fixtures to reference new package

remove unused message type

update header guard

rename message field in DynamicArrayPrimitivesNested

mikaelarguedas commented:
history of the files has been propagated, ready for review

dirk-thomas commented on test_msgs/package.xml:
```diff
@@ -0,0 +1,24 @@
+  <name>test_msgs</name>
+  <version>0.0.0</version>
+  <description>A package containing message definitions and fixtures used exclusively for testing purposes.</description>
+  <maintainer email="karsten@osrfoundation.org">Karsten Knese</maintainer>
```
Not sure why you picked @Karsten1987 but I don't mind :wink:

mikaelarguedas replied:
he is the one that started the original work in https://github.com/ros2/common_interfaces/pull/30 and I took over, I figured he'd like to keep maintainership :stuck_out_tongue_winking_eye:

dirk-thomas replied:
> I figured he'd like to keep maintainership

Very generous of you :laughing: Works for me...

dirk-thomas commented on test_msgs/CMakeLists.txt:
```diff
@@ -0,0 +1,53 @@
+)
+
+install(DIRECTORY src/test_msgs
+  DESTINATION "${PYTHON_INSTALL_DIR}/")
```
This Python package doesn't have an `__init__.py` file. Should it?

dirk-thomas replied:
Nvm, just found the file being added in a later commit.
